config-builder/src/main/java/com/ca/apim/gateway/cagatewayconfig/bundle/builder/BundleMetadataBuilder.java

```diff
@@ -52,16 +49,21 @@ public BundleMetadata build(final AnnotatedBundle annotatedBundle, final Bundle
             if (StringUtils.isNotBlank(projectInfo.getVersion())) {
                 name =  bundleName.substring(0, bundleName.indexOf(projectInfo.getVersion()) - 1);
             }
-            final String metadataId = StringUtils.isBlank(annotatedEntity.getMetadataId()) ? idGenerator.generate() :
-                    annotatedEntity.getMetadataId();
 
-            BundleMetadata.Builder builder = new BundleMetadata.Builder(encass.getType(), metadataId, name,
-                    projectInfo.getGroupName(), projectInfo.getVersion());
+            BundleMetadata.Builder builder = new BundleMetadata.Builder(encass.getType(), name, projectInfo.getName(), projectInfo.getGroupName(), projectInfo.getVersion());
             builder.description(annotatedEntity.getDescription());
-            builder.environmentEntities(getEnvironmentDependenciesMetadata(dependentEntities));
+            builder.referencedEntities(getEnvironmentDependenciesMetadata(dependentEntities));
+            if (!getEnvironmentDependenciesMetadata(dependentEntities).isEmpty()) {
```

Comment:
Assign it to variable instead of calculating environment entities twice

Reply:
done